Mage.Sets/src/mage/cards/i/IntrudeOnTheMind.java

```diff
@@ -0,0 +1,159 @@
+package mage.cards.i;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import mage.MageObject;
+import mage.abilities.Ability;
+import mage.abilities.effects.OneShotEffect;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.cards.Cards;
+import mage.cards.CardsImpl;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.Zone;
+import mage.counters.CounterType;
+import mage.filter.FilterCard;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.game.permanent.token.ThopterColorlessToken2;
+import mage.game.permanent.token.Token;
+import mage.players.Player;
+import mage.target.Target;
+import mage.target.TargetCard;
+import mage.target.common.TargetOpponent;
+
+/**
+ *
+ * @author DominionSpy
+ */
+public final class IntrudeOnTheMind extends CardImpl {
+
+    public IntrudeOnTheMind(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.INSTANT}, "{3}{U}{U}");
+
+        // Reveal the top five cards of your library and separate them into two piles. An opponent chooses one of those piles. Put that pile into your hand and the other into your graveyard.
+        // Create a 0/0 colorless Thopter artifact creature token with flying, then put a +1/+1 counter on it for each card put into your graveyard this way.
+        this.getSpellAbility().addEffect(new IntrudeOnTheMindEffect());
+    }
+
+    private IntrudeOnTheMind(final IntrudeOnTheMind card) {
+        super(card);
+    }
+
+    @Override
+    public IntrudeOnTheMind copy() {
+        return new IntrudeOnTheMind(this);
+    }
+}
+
+class IntrudeOnTheMindEffect extends OneShotEffect {
+
+    private static final FilterCard filter = new FilterCard("cards to put in the first pile");
+
+    IntrudeOnTheMindEffect() {
+        super(Outcome.DrawCard);
+        staticText = "Reveal the top five cards of your library and separate them into two piles. " +
+                "An opponent chooses one of those piles. Put that pile into your hand and the other into your graveyard. " +
+                "Create a 0/0 colorless Thopter artifact creature token with flying, " +
+                "then put a +1/+1 counter on it for each card put into your graveyard this way.";
```

Comment:
Wow cards these days have so much text. It's unfortunate this has to duplicate logic from the existing class but given the constraints this custom class does seem to be the way to go.